src/Orleans/Streams/PubSub/ImplicitStreamSubscriberTable.cs

```diff
@@ -163,57 +173,27 @@ private bool HasImplicitSubscription(string streamNamespace, int grainIdTypeCode
                 return false;
             }
 
-            HashSet<int> entry;
-            return (table.TryGetValue(streamNamespace, out entry) && // if we don't have implictit subscriptions for this namespace, fail out
-                    entry.Contains(grainIdTypeCode));                 // if we don't have an implicit subscription for this type of grain on this namespace, fail out
+            HashSet<int> entry = GetOrAddImplicitSubscribersSet(streamNamespace);
+            return entry.Contains(grainIdTypeCode);
         }
 
         /// <summary>
-        /// Add an implicit subscriber to the table.
+        /// Finds all implicit subscribed for the given stream namespace.
         /// </summary>
-        /// <param name="grainClass">Type of the grain class whose instances subscribe to the specified namespaces.</param>
-        /// <param name="namespaces">Namespaces instances of the grain class should subscribe to.</param>
-        /// <exception cref="System.ArgumentException">
-        /// No namespaces specified.
-        /// or
-        /// Duplicate specification of namespace "...".
-        /// </exception>
-        private void AddImplicitSubscriber(Type grainClass, ISet<string> namespaces)
+        /// <param name="streamNamespace">The stream namespace to find subscribers too.</param>
+        /// <returns></returns>
+        private HashSet<int> FindImplicitSubscribers(string streamNamespace)
```

Comment:
nit: This method is finding type code of implicit subscribers right? FindImplicitSubscriberTypeCode may be less confusing.